argusclient/client.py

```diff
@@ -391,7 +391,18 @@ def add(self, alert):
             alertobj.trigger = alertobj.triggers.add(alert.trigger)
         if alert.notification:
             alertobj.notification = alertobj.notifications.add(alert.notification)
-        if alert.trigger and alert.notification:
+        elif alert.notifications and isinstance(alert.notifications, list):
+            notifications = []
+            for notification in alert.notifications:
+                notifications.append(alertobj.notifications.add(notification))
+            alertobj.notifications = notifications
```

Comment:
Actually, I think this is not required now, since `alertobj.notifications.add()` is already taking care of populating the service cache, so you may just drop the local `notifications` `list`.